cluster/utils.sh

```diff
@@ -94,12 +94,28 @@ function create_cbt_table() {
     return 1
   fi
   if ! ( cbt -project ${PROJECT} createinstance ${CBT_INSTANCE} ${CBT_INSTANCE} \
-                                                ${CBT_INSTANCE}-c ${CBT_ZONE} 3 SSD && \
-         cbt -project ${PROJECT} -instance ${CBT_INSTANCE} createtable ${CBT_TABLE} && \
-         cbt -project ${PROJECT} -instance ${CBT_INSTANCE} createfamily ${CBT_TABLE} tfexample && \
-         cbt -project ${PROJECT} -instance ${CBT_INSTANCE} setgcpolicy ${CBT_TABLE} tfexample maxversions=1 &&
-         cbt -project ${PROJECT} -instance ${CBT_INSTANCE} createfamily ${CBT_TABLE} metadata && \
-         cbt -project ${PROJECT} -instance ${CBT_INSTANCE} setgcpolicy ${CBT_TABLE} metadata maxversions=1 ); then
+             ${CBT_INSTANCE}-c ${CBT_ZONE} 3 SSD ); then
+    echo "Could not create instance ${CBT_INSTANCE} in project ${PROJECT}"
+    return 1
+  fi
+}
+
+
+# Creates a Cloud Bigtable table for storing games.
+# Globals:
+#   PROJECT: The cloud project
+#   CBT_INSTANCE: The Cloud Bigtable instance within PROJECT (create if absent)
+#   CBT_TABLE:  The name of the Cloud Bigtable table to create in CBT_INSTANCE
+function create_cbt_table() {
+  check_cbt_exists
+  if ! ( cbt -project ${PROJECT} listinstances | grep -wq "^${CBT_INSTANCE}" ); then
+    create_cbt_instance
+  fi
+  if ! (cbt -project ${PROJECT} -instance ${CBT_INSTANCE} createtable ${CBT_TABLE} && \
+           cbt -project ${PROJECT} -instance ${CBT_INSTANCE} createfamily ${CBT_TABLE} tfexample && \
+           cbt -project ${PROJECT} -instance ${CBT_INSTANCE} setgcpolicy ${CBT_TABLE} tfexample maxversions=1 &&
```

Comment:
it's kind a weird that this doesn't have a \ and all the others do.

Reply:
It sure is! Yet it works, as I just reassured myself. Apparently bash will continue the line if it can tell that it hasn't terminated, as must be the case here because of an open parenthesis:

https://stackoverflow.com/questions/3871332/how-to-tell-bash-that-the-line-continues-on-the-next-line/35931689

More like Python than I had realized. It does make me want to *remove* the other backslashes, now! Seriously, I do agree that it should be consistent.